tsfc/tsfc_to_loopy.py

```diff
@@ -68,30 +70,45 @@ def index_to_iname(self, index):
 
         return iname
 
-    @staticmethod
-    def _is_cse_eligible(node):
+    def _is_cse_eligible(self, node):
+        if not (isinstance(node, g.Literal) and node.array.shape == ()):
```

Comment:
Easier to read with this condition inverted IMO.

Reply:
Agreed. Changed it.